### PR TITLE
add option to select backends TF/PT

### PR DESCRIPTION
**Pull request by thangckt**
add a new key in `param.json` file
```
"train_backend": "pytorch"/"tensorflow",
```
relate to this issue #1462

## Summary by CodeRabbit

- **New Features**
	- Enhanced model training and development processes.
	- Introduced new functionalities for model deviation analysis with various system configurations.

### CODE REVIEW

File: dpgen/generator/run.py [hunk 1]
```diff
@@ -125,6 +125,16 @@
 run_opt_file = os.path.join(ROOT_PATH, "generator/lib/calypso_run_opt.py")
 
 
+def _get_model_suffix(jdata) -> str:
+    """Return the model suffix based on the backend"""
+    backend = jdata.get("train_backend", "tensorflow")
+    if backend == "tensorflow":
+        suffix = ".pb"
+    elif backend == "pytorch":
+        suffix = ".pth"
```
Comment (njzjz) [inline dpgen/generator/run.py:134]: should use `else` to handle other cases.

File: dpgen/generator/run.py [hunk 2]
```diff
@@ -125,6 +125,16 @@
 run_opt_file = os.path.join(ROOT_PATH, "generator/lib/calypso_run_opt.py")
 
 
+def _get_model_suffix(jdata) -> str:
+    """Return the model suffix based on the backend"""
+    backend = jdata.get("train_backend", "tensorflow")
```
Comment (njzjz) [inline dpgen/generator/run.py:130]: This argument should be added into `arginfo.py` for documentation and check.

File: dpgen/generator/run.py [hunk 3]
```diff
@@ -799,17 +814,23 @@ def run_train(iter_index, jdata, mdata):
...
-    if jdata.get("dp_compress", False):
-        backward_files.append("frozen_model_compressed.pb")
+    if suffix == ".pb":
+        backward_files += [
+            "model.ckpt.meta",
+            "model.ckpt.index",
+            "model.ckpt.data-00000-of-00001",
+        ]
+        if jdata.get("dp_compress", False):
+            backward_files.append("frozen_model_compressed%s" % suffix)
```
Comment (njzjz) [inline dpgen/generator/run.py:832]: should handle the situation for pt.

For pt, the checkpoint file is `model.pt`; the compressed model file will be `frozen_model_compressed.pth`

File: dpgen/generator/run.py [hunk 4]
```diff
@@ -762,9 +777,9 @@ def run_train(iter_index, jdata, mdata):
         if training_init_model:
             init_flag = " --init-model old/model.ckpt"
         elif training_init_frozen_model is not None:
-            init_flag = " --init-frz-model old/init.pb"
+            init_flag = " --init-frz-model old/init%s" % suffix
         elif training_finetune_model is not None:
-            init_flag = " --finetune old/init.pb"
+            init_flag = " --finetune old/init%s" % suffix
         command = f"{train_command} train {train_input_file}{extra_flags}"
```
Comment (njzjz) [inline dpgen/generator/run.py:783]: For pt, we use `dp --pt train`

File: dpgen/generator/run.py [hunk 5]
```diff
@@ -880,13 +901,15 @@ def post_train(iter_index, jdata, mdata):
...
         return
     # symlink models
+    suffix = _get_model_suffix(jdata)
     for ii in range(numb_models):
-        if not jdata.get("dp_compress", False):
-            model_name = "frozen_model.pb"
-        else:
-            model_name = "frozen_model_compressed.pb"
+        model_name = "frozen_model%s" % suffix
+        if suffix == ".pb":
```
Comment (njzjz) [inline dpgen/generator/run.py:907]: should preserve the situation for pt. I think tf and pt share the same logic here.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpgen/generator/run.py (hunk 4 above)
Comment (thangckt) [reply dpgen/generator/run.py:783]: I think this should be set from the key: 
```json
"train": [
        {
	        "command": "dp --pt"
        }
]
```

File: dpgen/generator/run.py (hunk 3 above)
Comment (njzjz) [reply dpgen/generator/run.py]: It should be `model.ckpt.pt`

File: dpgen/generator/run.py (hunk 4 above)
Comment (njzjz) [reply dpgen/generator/run.py:787]: This is okay, but needs documentation.

File: dpgen/generator/run.py [hunk 6]
```diff
@@ -3770,6 +3838,8 @@ def make_fp_calculation(iter_index, jdata, mdata):
         make_fp_amber_diff(iter_index, jdata)
     elif fp_style == "custom":
         make_fp_custom(iter_index, jdata)
+    elif fp_style == "gpaw":
```
Comment (njzjz) [inline dpgen/generator/run.py:3841]: Could you submit gpaw in a different PR?

**Pull request closed without merging**